modules/gdscript/gdscript_parser.cpp

```diff
@@ -3582,6 +3583,31 @@ bool GDScriptParser::export_annotations(const AnnotationNode *p_annotation, Node
 			variable->export_info.hint_string = hint_prefix + ":" + variable->export_info.hint_string;
 			variable->export_info.type = Variant::ARRAY;
 		}
+
+		if (p_annotation->name == SNAME("@export_action")) {
+			variable->export_info.hint_string = "action";
+			if (!p_annotation->resolved_arguments.is_empty()) { // Icon Name
+				variable->export_info.hint_string += (String)p_annotation->resolved_arguments[0];
+			}
+			variable->export_info.usage = PROPERTY_USAGE_EDITOR;
+			if (variable->export_info.type == Variant::Type::CALLABLE) {
+				if (!this->is_tool()) {
+					push_error(R"(Exported actions only work in tool scripts.)", p_annotation);
+				}
+			} else {
+				push_error(R"(Exported actions can only be of type Callable.)", p_annotation);
+			}
```

Comment:
```suggestion
			if (variable->export_info.type != Variant::Type::CALLABLE) {
				push_error(R"(Exported actions can only be of type Callable.)", p_annotation);
			} else if (!is_tool()) {
				push_error(R"(Exported actions only work in tool scripts.)", p_annotation);
			}
```
No point in nesting this.